File: src/main/java/endytkn/randomEvents/CommandRegister.java
package endytkn.randomEvents;

import endytkn.randomEvents.randomEvent.RandomEventCommands;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.RegisterCommandsEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;
import net.minecraftforge.fml.common.Mod;

@Mod.EventBusSubscriber
public class CommandRegister {
    static {
        MinecraftForge.EVENT_BUS.register(CommandRegister.class);
    }

    @SubscribeEvent
    public static void onRegisterCommands(RegisterCommandsEvent event) {
        RandomEventCommands.register(event.getDispatcher());
    }
}

File: src/main/kotlin/endytkn/randomEvents/CommandRegister.kt
package endytkn.randomEvents.events

import endytkn.randomEvents.commands.RandomEventsCommands
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.event.RegisterCommandsEvent
import net.minecraftforge.event.TickEvent
import net.minecraftforge.eventbus.api.SubscribeEvent
import net.minecraftforge.fml.common.Mod
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger

@Mod.EventBusSubscriber
object CommandRegister {
    private val LOGGER: Logger = LogManager.getLogger()

    init {
        MinecraftForge.EVENT_BUS.register(this)
    }

    @SubscribeEvent
    fun onRegisterCommands(event: RegisterCommandsEvent) {
        RandomEventsCommands.register(event.dispatcher)
    }

    @SubscribeEvent
    fun onTickPlayer(event: TickEvent.PlayerTickEvent) {
        val player = event.player
        val level = event.player.level()
    }
}
